tee-worker/app-libs/stf/src/trusted_call.rs

```diff
@@ -745,41 +686,40 @@ where
 				let account = SgxParentchainTypeConverter::convert(
 					who.to_account_id().ok_or(Self::Error::InvalidAccount)?,
 				);
-				let call_index =
-					node_metadata_repo.get_from_metadata(|m| m.vc_issued_call_indexes())??;
 
-				let key = Self::request_vc_callback_internal(
+				Self::request_vc_callback_internal(
 					signer.to_account_id().ok_or(Self::Error::InvalidAccount)?,
-					who,
 					assertion.clone(),
 				)
 				.map_err(|e| {
 					debug!("pushing error event ... error: {}", e);
 					add_call_from_vcmp_error(
 						calls,
-						node_metadata_repo,
+						node_metadata_repo.clone(),
 						Some(account.clone()),
 						e.to_vcmp_error(),
 						hash,
 					);
 					e
 				})?;
 
+				debug!("pushing vc_issued event ...");
+				let call_index =
+					node_metadata_repo.get_from_metadata(|m| m.vc_issued_call_indexes())??;
+
 				calls.push(OpaqueCall::from_tuple(&(
-					call_index,
-					account,
-					assertion,
-					vc_index,
-					vc_hash,
-					aes_encrypt_default(&key, &vc_payload),
-					hash,
+					call_index, account, assertion, vc_index, vc_hash, hash,
 				)));
-				let res = RequestVCResult {
-					vc_index,
-					vc_hash,
-					vc_payload: aes_encrypt_default(&key, &vc_payload),
-				};
-				Ok(TrustedCallResult::RequestVC(res))
+
+				if let Some(key) = maybe_key {
+					Ok(TrustedCallResult::RequestVC(RequestVCResult {
+						vc_index,
+						vc_hash,
+						vc_payload: aes_encrypt_default(&key, &vc_payload),
+					}))
+				} else {
+					Ok(TrustedCallResult::Empty)
```

Comment:
I just have a question here, Isn't it wasteful to process a `request-vc` request, when `maybe_key` is `None` since the Result returned is empty. I might be missing something here, Let me know what you thing.

Reply:
Yes, it makes little sense to request a vc via parachain now, as you simply don't get the vc content 😂 

as mentioned in the PR:
> VC-request via II will be gradually phase out, you could theoretically still request it via parachain extrinsic, but it makes less sense if the generated vc can't be parsed